lib/public/views/Runs/RunPerDataPass/RunsPerDataPassOverviewModel.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * @license
+ * Copyright CERN and copyright holders of ALICE O2. This software is
+ * distributed under the terms of the GNU General Public License v3 (GPL
+ * Version 3), copied verbatim in the file "COPYING".
+ *
+ * See http://alice-o2.web.cern.ch/license for full licensing information.
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+import { RemoteData } from '/js/src/index.js';
+import { buildUrl } from '../../../utilities/fetch/buildUrl.js';
+import { detectorsProvider } from '../../../services/detectors/detectorsProvider.js';
+import { RunsOverviewModel } from '../Overview/RunsOverviewModel.js';
+import { ObservableData } from '../../../utilities/ObservableData.js';
+import { getRemoteDataSlice } from '../../../utilities/fetch/getRemoteDataSlice.js';
+
+/**
+ * Runs Per Data Pass overview model
+ */
+export class RunsPerDataPassOverviewModel extends RunsOverviewModel {
+    /**
+     * Constructor
+     * @param {Model} model global model
+     */
+    constructor(model) {
+        super(model);
+        this._detectors = new ObservableData(RemoteData.notAsked());
+        this._detectors.bubbleTo(this);
+        this._dataPass = new ObservableData(RemoteData.notAsked());
+        this._dataPass.bubbleTo(this);
+    }
+
+    /**
+     * Retrieve a list of detector types from detectorsProvider
+     *
+     * @return {Promise<void>} resolves once the data has been fetched
+     * @private
+     */
+    async _fetchDetectors() {
+        this._detectors.setCurrent(RemoteData.loading());
+        try {
+            this._detectors.setCurrent(RemoteData.success(await detectorsProvider.getAll()));
+        } catch (error) {
+            this._detectors.setCurrent(RemoteData.failure(error));
+        }
+    }
+
+    /**
+     * Fetch data pass data which runs are fetched
+     * @return {Promise<void>} promise
+     */
+    async _fetchDataPass() {
+        try {
+            const { items: [dataPass] = [] } = await getRemoteDataSlice(`/api/dataPasses?filter[ids][]=${this._dataPassId}`);
+            this._dataPass.setCurrent(dataPass);
+        } catch (error) {
+            this._dataPass.setCurrent(RemoteData.failure(error));
+        }
+    }
+
+    // eslint-disable-next-line valid-jsdoc
+    /**
+     * @inheritdoc
+     */
+    async load({ dataPassId } = {}) {
+        this._dataPassId = dataPassId || this._dataPassId;
+        if (!this._dataPassId) {
+            return;
+        }
+        this._fetchDetectors();
+        this._fetchDataPass();
+        super.load();
+    }
```

Comment:
I think it is a bad idea to change the signature of inherited methods. I know here you respect the contravariance, but still: you are using a parent function for very different context. In the abstraction, it suppose that your model is configured, you should refrain from the shortcut of configuring & loading at the same time.
Here the function is named load, it should not accept a parameter to set the data pass id. Instead, you should have a `setDataPassId` that call load.

In any case, when you change the signature, you can't rely on `inheritDoc`! Here the dataPassId parameter is documented nowhere.

Reply:
applied